docs/examples/perceptualdecisionmaking.ipynb

```diff
@@ -0,0 +1,434 @@
+    "\n",
+    "- Tracking and saving behavioral data (rewards, actions, observations) every `sv_per` steps.\n",
+    "- Generating visualization figures during training if` sv_fig=True`.\n",
+    "- Providing progress information if `verbose=True`.\n",
+    "\n",
+    "Here, we’ll use the wrapper solely to compute the agent’s performance, but later it will help us assess learning and save intermediate results.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualize example trials\n",
+    "fig = ngym.utils.plot_env(\n",
+    "    env,\n",
+    "    name='Perceptual Decision Making',\n",
+    "    ob_traces=[\n",
+    "        'Fixation',\n",
+    "        'Stim 1',\n",
+    "        'Stim 2',\n",
+    "        'PassReward', # Reward for the previous action\n",
+    "        'PassAction' # Action taken in the previous step\n",
+    "    ],\n",
+    "    num_trials=5,\n",
+    ")\n",
+    "\n",
+    "# Evaluate performance of the environment before training\n",
+    "eval_monitor = Monitor(\n",
+    "    env\n",
+    ")\n",
+    "print(\"\\nEvaluating random policy performance...\")\n",
+    "metrics = eval_monitor.evaluate_policy(num_trials=EVAL_TRIALS)\n",
+    "print(f\"\\nRandom policy metrics ({EVAL_TRIALS:,} trials):\")\n",
+    "print(f\"Mean performance: {metrics['mean_performance']:.4f}\")\n",
+    "print(f\"Mean reward: {metrics['mean_reward']:.4f}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see, the agent's behavior is entirely random. Through training, we expect the agent to improve by learning to respect the fixation period, and map signal peaks to the correct choices in the ring representation. Let’s move on to training the agent to see whether it can learn these key aspects of the task.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1.2 Training and Evaluating the Agent\n",
+    "\n",
+    "We will now train the agent using Stable-Baselines3’s implementation of [PPO (Proximal Policy Optimization)](https://stable-baselines3.readthedocs.io/en/master/modules/ppo.html), a widely used reinforcement learning algorithm known for its stability and efficiency.\n",
+    "\n",
+    "To support recurrent policies, we will use [RecurrentPPO](https://sb3-contrib.readthedocs.io/en/master/modules/ppo_recurrent.html#recurrent-ppo), which extends PPO with recurrent neural networks, specifically LSTMs.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.2.1 Training the Agent\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the number of trials to train on\n",
+    "avg_timesteps = 7 # Observed\n",
+    "TRAIN_TRIALS = 10000  # Choose the desired number of trials\n",
```

Comment:
If we want to go ahead with #217 , please decrease this number to about 500 to avoid adding 5 minutes to the notebook workflow, and add a comment asking users to increase it again. I did this for the other notebooks like [this](f20632f).